### DISCUSSION TITLE
How do I store my own config in source control?

### DISCUSSION THREAD

**Discussion by tonekk** [NvChad/NvChad#1426] [Q&A] [answered]
Hi,

first off: Nice project, love it :)

From what I understand, I should not create a fork to configure my own version of NvChad.
What I don't understand though, is how should I keep my own config in my github so that the setup is a oneliner? Any ideas?

EDIT: My current solution is to use a fork and add the `lua/custom` folder as a submodule. This shouldn't cause conflicts, since `lua/custom` is in the `.gitignore` file. So I just have to sync the fork before calling `NvChadUpdate`. This is surely not how it was intended, but it works well with my current dotfiles setup:
https://github.com/tonekk/nvim/tree/main/lua

**Comment by siduck** [comment, marked as answer]
@tonekk for the time being i just put the custom dir to my dotfiles repo. soon @LeonHeidelbach is going to improve our snapshot feature so you can store your whole nvchad config ( default config + custom config + all plugins code ) into a single minified file so its easy to rollback to your preferred snapshot i.e for stability and then you can save that snapshot file in git etc, its portable

**Reply by tonekk** [reply]
Thanks for the fast reply 💯

**Reply by simon-liebehenschel** [reply]
> soon @LeonHeidelbach is going to improve our snapshot feature so you can store your whole nvchad config ( default config + custom config + all plugins code ) into a single minified file

Hello. It is 2023 now. Did this happen?

**Reply by siduck** [reply]
hi @AIGeneratedUsername no it did before but was using packer stuff. I dont recall why i removed it. But i dont think users would need snapshot now as nvchad doesnt break anymore

**Reply by AlexRipoll** [reply]
Hi @siduck. So then, what is the recomended procees for pushing the custom folder to another git remote?

**Reply by siduck** [reply]
> Hi @siduck. So then, what is the recomended procees for pushing the custom folder to another git remote?

Just clone your dotfiles repo somewhere, for example

~/dotfiles/nvchad

Then symlink it 

ln -s ~/dotfiles/nvchad ~/.config/nvim/lua/custom 

I haven't tried it but it should work.

**Reply by uxduck** [reply]
The symlink approach works... but

I needed to edit the `chadrc.lua` file and save it, otherwise the config that I symlink was not used/respected by neovim/NvChad, even after restarting the NvChad and terminal. 

- I've got my nvim custom config in https://github.com/ben-xD/dotfiles-macos/tree/main/.config/dotfiles/nvim-custom
- And I sym link in my setup.sh script: https://github.com/ben-xD/dotfiles-macos/blob/22fce9d9e319c249ccd9a8d10f3211783e6ea33c/.config/dotfiles/setup.sh#L40

**Question:** Is there a bash command I could run to reload the config after symlinking?

**Reply by siduck** [reply]
@ben-xD nvchad custom config reloads when you save any file in custom dir. Why would you want to reload the config if you dont have any changes?

**Reply by uxduck** [reply]
I do technically make "changes" when I run `ln -s` (symlink) files from my dotfiles repo. Neovim/NvChad doesn't know about the new, symlinked config.

**Reply by siduck** [reply]
@ben-xD  i tried it here and saving the file for example `~/dotfiles/nvchad_custom` , it reloads . Idk why it doesnt work for u! 

btw do know that nvchad stuff only reloads when you write the files with nvim

**Reply by uxduck** [reply]
Saving the file in nvim works for me too. But when I use an existing config from another machine (for example, after cloning a dotfiles repo and symlinking the custom folder), nvim doesn't yet notice the old config file is gone, and the new one is symlinked.

**Reply by siduck** [reply]
check the autocmd in core/init.lua, its used for reloading stuff

just copy that and make a custom function and then use nvim --headless to call it

**Comment by saumyajyoti** [comment]
I keep Custom folder in my own repo and then symlink it to NVChad/lua folder.

**Comment by danielo515** [comment]
You can also use chezmoi. It is a manager that keeps your dotfiles separated from their final location and it just copies them to the destination when needed.
